gnosis/safe/account_abstraction/safe_operation.py

```diff
@@ -0,0 +1,140 @@
+import dataclasses
+from typing import Optional
+
+from eth_abi import encode as abi_encode
+from eth_abi.packed import encode_packed
+from eth_typing import ChecksumAddress
+from hexbytes import HexBytes
+
+from gnosis.eth.account_abstraction import UserOperation
+from gnosis.eth.utils import fast_keccak
+
+_domain_separator_cache = {}
+
+
+@dataclasses.dataclass
+class SafeOperation:
```

Comment:
As far as I know SafeOperation is adding some atributes to userOperation, could be this an extension of UserOperation class?

Reply:
Not sure about it, some have different names so I will keep it like this until there's a final version, but good point